controllers/azuremanagedcontrolplane_controller.go

```diff
@@ -342,3 +342,5 @@ func (amcpr *AzureManagedControlPlaneReconciler) ClusterToAzureManagedControlPla
 
 	return nil
 }
+
+var getNewAzureManagedControlPlaneReconciler = newAzureManagedControlPlaneReconciler
```

Comment:
In case we ever run tests in parallel that each modify this function, I could see having a single variable like this causing race conditions. Is there a way we can scope these implementation details to an individual instance of something like `AzureManagedControlPlaneReconciler` like we do for the AzureMachine tests?

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/454d632e21446c5796797bb8ff03977929ecb7e0/controllers/azuremachine_reconciler.go#L107-L109

Reply:
I was thinking about this and maybe deferring it back to the actual implementation however, that doesn't allow us being able to catch that when parralell uses this package level variable to "patch".  

If we make this part of the actual reciever 
```go
func (amcpr *AzureManagedControlPlaneReconciler) getNewAzureManagedControlPlaneReconciler = newAzureManagedControlPlaneReconciler
```
we may be in better shape which mimics what you have above.  I can make these changes.

Reply:
Instead of making it part of the reciever, I took the liberty to do the following: 
```go
type AzureManagedControlPlaneReconciler struct {
     client.Client
     Recorder                                 record.EventRecorder
     ReconcileTimeout                         time.Duration
     WatchFilterValue                         string
     getNewAzureManagedControlPlaneReconciler func(scope *scope.ManagedControlPlaneScope) (*azureManagedControlPlaneService, error)
   }
```
On instantiation it gets set to: 

```go
amcpr.getNewAzureManagedControlPlaneReconciler = newAzureManagedControlPlaneReconciler
```

Testing allows us to rescope this as demonstrated in this PR.

Reply:
SGTM, thanks!